### PR TITLE
add plugins catalog

### LINKED ISSUE #50: Add plugin catalog support
#### Why should this feature be added?

The plugin catalog will allow us to get info about plugin compatibility, etc.

#### How would you like the feature to work?

`csctl get plugin-catalog`, plus a clean API on the client side

#### What alternatives have been considered?
N/A

#### Additional Context
N/A

### PR DESCRIPTION
**Pull request by ashleyschuett**
### Description

This PR adds listing the plugin catalog which lists available plugins for clusters

 #### What does this pull request accomplish?

Adds listing the plugin catalog. 

 #### What issue(s) does this fix?
* Fix #50 

 #### Additional Considerations

- Need to add filtering right now it just lists all plugins. 

- If we add a new type of plugin we will have to update the cli, which seems not great 

 ### Testing

```
$ csctl get plugc
TYPE				IMPLEMENTATION		VERSIONS	KUBERNETES_COMPATIBILITY
Cloud Controller Manger		azure			1.0.0		1.10.x - 1.12.x
Cloud Controller Manger		amazon_web_services	1.0.0		1.10.x - 1.12.x
Cloud Controller Manger		digitalocean		0.1.7		1.11.x - 1.12.x
Cloud Controller Manger		digitalocean		0.1.6		1.10.x - 1.11.x
Cloud Controller Manger		google_cloud_platform	2.0.0		1.12.x - 1.12.x
Cloud Controller Manger		google_cloud_platform	1.0.0		1.10.x - 1.11.x
Cloud Controller Manger		packet			0.0.4		1.10.x - 1.12.x
Cluster Management		containership		4.0.1		1.10.x - 1.12.x
Cluster Management		containership		3.0.1		1.9.x - 1.11.x
Cluster Management		containership		3.0.0		1.9.x - 1.11.x
Cluster Management		containership		2.1.0		1.8.x - 1.10.x
Cluster Management		containership		2.0.1		1.8.x - 1.10.x
Cluster Management		containership		2.0.0		1.8.x - 1.10.x
Cluster Management		containership		1.0.3		1.7.x - 1.9.x
CNI				calico			2.0.0		1.11.x - 1.12.x
CNI				calico			1.0.0		1.10.x - 1.10.x
CNI				canal			2.0.0		1.10.x - 1.12.x
CSI				azure			0.1.0		1.10.x - 1.11.x
CSI				digitalocean		0.3.0		1.12.x - 1.12.x
CSI				digitalocean		0.2.0		1.10.5 - 1.11.x
CSI				digitalocean		0.1.1		1.10.x - 1.11.x
CSI				packet			0.3.0		1.10.x - 1.11.x
CSI				google_cloud_platform	0.2.0-alpha	1.12.x - 1.12.x
CSI				google_cloud_platform	0.1.1		1.10.x - 1.11.x
CSI				google_cloud_platform	0.1.0		1.10.x - 1.11.x
Logs				kubernetes		1.0.0		1.x.x - 1.x.x
Metrics				prometheus		1.0.0		1.7.x - 1.12.x
```

 #### Setup

`make install`

 #### Instructions

run `csctl get plugin-catalog`

### CODE REVIEW

**Comment by mattkelly** [comment]
I did end up making a ticket for this just FYI: https://github.com/containership/csctl/issues/50

File: cloud/api/swagger.yaml [hunk 1]
```diff
@@ -427,6 +518,121 @@ definitions:
         title: Plugin configuration (varies per-plugin)
         type: object
 
+  PluginCatalog:
+    title: A Containership plugin definition
+    type: object
+    required:
+      - cloud_controller_manager
+      - cluster_management
+      - cni
```
Comment (mattkelly) [inline cloud/api/swagger.yaml]: I now see that this was capitalized before in order to properly capitalize the generated go types. Can you make it all caps again? My bad :(

Comment (ashleyschuett) [reply cloud/api/swagger.yaml]: To be fair I hate both ways.

File: cloud/api/swagger.yaml [hunk 2]
```diff
@@ -427,6 +518,121 @@ definitions:
         title: Plugin configuration (varies per-plugin)
         type: object
 
+  PluginCatalog:
+    title: A Containership plugin deginition
+    type: object
+    required:
+      - cloud_controller_manager
+      - cluster_management
+      - CNI
```
[the remark this answers is not in the document]
Comment (mattkelly) [reply cloud/api/swagger.yaml]: Lowercase was the wrong answer due to the way go types are named when generated.

**Pull request merged**